resources/assets/js/forms.js

```diff
@@ -287,72 +273,75 @@ ready(() => {
-                <div class="modal-header">
-                  <h5 class="modal-title">${element.dataset.confirm_submit_title ?? ''}</h5>
-                  <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
-                </div>
-                <div class="modal-body${element.dataset.confirm_submit_text ? '' : ' d-none'}">
-                  <p>${element.dataset.confirm_submit_text ?? ''}</p>
-                </div>
-                <div class="modal-footer">
-                  <button type="button" class="${element.className}"
-                    autofocus
-                    title="${element.title}" data-submit="">
-                    ${element.dataset.confirm_button_text ?? element.innerHTML}
-                  </button>
-                </div>
+    event.preventDefault();
+
+    document.getElementById('confirmation-modal')?.remove();
+    document.body.insertAdjacentHTML(
+      'beforeend',
+      `
+        <div class="modal" tabindex="-1" id="confirmation-modal">
+          <div class="modal-dialog">
+            <div class="modal-content ${document.body.dataset.theme_type === 'light' ? 'bg-white' : 'bg-dark'}">
+              <div class="modal-header">
+                <h5 class="modal-title">${element.dataset.confirm_submit_title ?? ''}</h5>
+                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+              </div>
+              <div class="modal-body${element.dataset.confirm_submit_text ? '' : ' d-none'}">
+                <p>${element.dataset.confirm_submit_text ?? ''}</p>
+              </div>
+              <div class="modal-footer">
+                <button type="button" class="${element.className}"
+                  autofocus
+                  title="${element.title}" data-submit="">
+                  ${element.dataset.confirm_button_text ?? element.innerHTML}
+                </button>
               </div>
             </div>
           </div>
-        `
-      );
-
-      const modal = document.getElementById('confirmation-modal');
-      modal.addEventListener('hide.bs.modal', () => {
-        modalOpen = false;
-      });
-
-      const modalSubmitButton = modal.querySelector('[data-submit]');
-      modalSubmitButton.addEventListener('click', () => {
-        element.type = oldType;
-        element.click();
-      });
+        </div>
+      `
+    );
+
+    const modal = document.getElementById('confirmation-modal');
+    let bootstrapModal = new bootstrap.Modal(modal);
+    modal.addEventListener('hide.bs.modal', () => {
+      modalOpen = false;
+    });
 
-      /**
-       * After the modal has been shown, focus on the "Submit" button in the modal
-       * so that it can be confirmed with "Enter".
-       */
-      modal.addEventListener('shown.bs.modal', () => {
-        modalSubmitButton.focus();
-      });
+    const modalSubmitButton = modal.querySelector('[data-submit]');
```

Comment:
I really am a big fan of element IDs. This is always fuzzy and can maybe become a problem.

Reply:
In this case there are many forms (as in: number of angels availabe) so any ID would not be feasible